CONTRIBUTING.md

```diff
@@ -0,0 +1,124 @@
+
+### External contributors
+
+If you are not a member of the `snowplow` GitHub organization, you can contribute to `dbt-snowplow-*` by forking the relevant package repository. For a detailed overview on forking, check out the [GitHub docs on forking](https://help.github.com/en/articles/fork-a-repo). In short, you will need to:
+
+1. Fork this repository
+2. Clone your fork locally
+3. Check out a new branch for your proposed changes
+4. Push changes to your fork
+5. Open a Pull Request against this repo from your forked repository
+
+### Snowplow contributors
+
+If you are a member of the `snowplow` GitHub organization, you will have push access to this repo. Rather than forking to make your changes, just clone the repository, check out a new branch, and push directly to that branch.
+
+## Setting up an environment
+
+Assuming you already have dbt installed, it will be beneficial to create a profile for any warehouse connections you have when it comes to testing the changes to your package. The easiest way to do this that will involve the least changes to the testing setup is to create an `integration_tests` profile and populate it with any connections you have to our supported warehouse types (redshift+postgres, databricks, snowflake, bigquery). 
+
+**It is recommended you use a custom schema for integration tests.**
+
+```yml
+integration_tests:
+  outputs:
+    databricks:
+      type: databricks
+      ...
+    snowflake:
+      type: snowflake
+      ...
+    bigquery:
+      type: bigquery
+      ...
+    redshift:
+      type: redshift
+      ...
+    postgres:
+      type: postgres
+      ...
+  target: postgres
+```
+
+## Implementation guidelines
+
+In general we try to follow these rules of thumb, but there are possible exceptions:
+- Dispatch any macro where it needs to support multiple warehouses. 
+  - Use inheritance where possible i.e. only define a macro for `redshift` if it is different to `postgres`, the same for `databricks` and `spark`
+- Where models need to be different across multiple warehouse types, ensure they are enabled based on the `target.type`
+- Make use of macros (ours and dbt's) where possible to avoid duplication and to manage the differences between warehouses
+  - Do not reinvent the wheel e.g. make use of [`type_*` macros](https://docs.getdbt.com/reference/dbt-jinja-functions/cross-database-macros#data-type-functions) instead of explicit datatypes
+  - In the case where a macro may be useful outside of a specific package, we may make the choice to add it to `dbt-snowplow-utils` [repository](https://www.github.com/snowplow/dbt-snowplow-utils) instead
+- Make use of the incremental logic as much as possible to avoid full-scanning large tables
+- Where new functionality is being added, or you are touching existing functionality that does not have good/any test, add tests
+
+## Testing
+
```

Comment:
I would probably highlight that the current integration testing suite does not cover the data models produced by the python scripts (we should probably want to add Great Expectations or sth similar at some point if it gets changed) but that they are still welcome to recommend code changes just preferably with suggested GE test scenario(s) to test it against?

Reply:
And that they should suggest the changes in the python scripts rather than sharing a new docker image. Cause this way we could see the changes in git I would think but not with the other.